examples/metalk8s-solution-example/operator/go.mod

```diff
@@ -1,10 +1,11 @@
-module example-operator
+module example-solution-operator
 
 require (
 	github.com/NYTimes/gziphandler v1.0.1 // indirect
 	github.com/go-openapi/spec v0.19.0
 	github.com/operator-framework/operator-sdk v0.9.1-0.20190805223000-66e78cc576ef
 	github.com/spf13/pflag v1.0.3
+	gopkg.in/yaml.v2 v2.2.2
```

Comment:
Did you look into [`spf13/viper`](https://github.com/spf13/viper)?

Reply:
It supports a bunch of features out-of-the-box, such as live reloading of config files, that would help us a lot.

Reply:
No, didn't check this, I will.

Reply:
Why would you want live-reloading of configs? :-/

Reply:
Importing a new Solution version would immediately allow Operators to pick it up if we propagate the repositories in their ConfigMaps.

Reply:
But that operator will not know about said version (or how to deploy it). So a restart would be required anyway.

In general, an app which starts, loads config, logs it and then uses that (static) config until it gets restarted is much easier to troubleshoot than some app which can dynamically at random points in time change its config, IMHO.

Also, operators are (or should be) trivial to restart (that's the whole point of stateless control loops).

Reply:
OK, then maybe we don't need those extra features.

I was looking at `prometheus-operator`, and they seem to use the standard [`flag`](https://golang.org/pkg/flag/) package: see [here](https://github.com/coreos/prometheus-operator/blob/3a86bb3115d59db459c92fb54ea63248ee4e4d3a/cmd/operator/main.go#L124-L158)
No way to pass a config file here.

`kubectl` uses `spf13/cobra` and some complex patterns to load its kubeconfig (tried to read through the code but it's confusing for a noob like me), registering all config options with command-line flag overrides.

Since Operator SDK sets up the manager with `spf13/pflag`, it's not too far from how `spf13/viper` works (e.g. see [this PR](https://github.com/operator-framework/operator-sdk/pull/950/files) converting from one to the other). Maybe we don't care about live-reload, but having a config-file with command-line flags overrides is nice for testing or simpler setups.
We can build our own in-house parsing of such files, but it seems the ecosystem often relies on `viper` and `cobra` for Go CLIs, so I wouldn't worry too much about it and just follow the trend.

Also, not sure if there's a standard way of validating an `apiVersion`/`kind` tuple in the tooling controllers rely on, but I'd assume there is...

Reply:
(of course, all those changes can be considered in a future PR! we just need to think about "real" Operators that will be built around this tooling, and the closer we can get to this with the example, the more likely we'll define better tooling)